File: python/longest_consecutive_sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        numSet = set(nums)

        for n in numSet:
            if n - 1 in numSet:
                continue
            
            cur = 1
            while n + 1 in numSet:
                cur += 1
                n += 1

            res = max(cur, res)
        return res

File: C++/longest_consecutive_sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int res = 0;
        unordered_set<int> numSet(nums.begin(), nums.end());

        for (auto& n : numSet){
            if (numSet.contains(n - 1)){
                continue;
            }

            int cur = 1;
            int i = n;
            while (numSet.contains(i + 1)){
                i++;
                cur++;
            }

            res = max(res, cur);
        }
        return res;
    }
};
